rfid2/rfid2.ino

```diff
@@ -0,0 +1,85 @@
+#include <Keyboard.h>
+#include <SPI.h>
+#include <MFRC522.h>
+
+#define SS_PIN 10
+#define RST_PIN 5
+#define KEY_RETURN 0xB0  // The hex value for the return key is 0xB0.
+
+MFRC522 mfrc522(SS_PIN, RST_PIN);
+char Enter = KEY_RETURN;  // Return key is declared as Enter.
+String readid;
+const String cards[] = {
+  "3A3D9D9E",
+  "3AF3BE9E",
+  "E222CF1D"
+  };  // Change this value to the UID of your card.
+int numberOfCards = (sizeof(cards) / sizeof(cards[0]));
+bool screenLocked = false;
+
+void setup() {
+  Serial.begin(9600);
+  Keyboard.begin();
+  SPI.begin();
+  mfrc522.PCD_Init();
+  Serial.println("Setup complete. Scan your RFID tag.");
+}
+
+void temp(byte *buffer, byte bufferSize) {
+  // Function to store card UID as a string datatype.
+  readid = "";
+  for (byte i = 0; i < bufferSize; i++) {
+    readid = readid + String(buffer[i], HEX);
+  }
+  readid.toUpperCase(); // Ensure the UID is in uppercase
+  Serial.print("Card UID: ");
+  Serial.println(readid);
+}
+
+void lockScreen() {
+  Serial.println("Card recognized. Locking screen.");
+  Keyboard.press(KEY_LEFT_GUI);  // Press the left Windows key.
+  Keyboard.press('l');           // Press the "l" key.
+  Keyboard.releaseAll();
+  delay(200);         // Release all keys.
+  screenLocked = true;
+}
+
+void unlockScreen() {
+  Serial.println("Card recognized. Unlocking screen.");
+  Keyboard.print("Cabinet24");       // Change this value to your Windows PIN/Password.
+  Keyboard.press(Enter);         // Press the Enter key to submit the PIN/Password
+  Keyboard.releaseAll();
+  delay(200);
+  screenLocked = false;
+}
+
+bool isCardValid(String cardID) {
+  for(int i=0; i < numberOfCards; i++) {
+    if(cardID == cards[i]) {
+      return true;
+    }
+  }
+  return false;
+}
+
+void loop() {
+  if (!mfrc522.PICC_IsNewCardPresent()) {
```

Comment:
Infinite loops are not good practice.

Reply:
Thanks for your feedback. This behavior is necessary for continuously monitoring sensors and handling events in real-time applications, such as reading RFID tags. Removing the infinite loop here would halt the program, making it non-functional for its intended purpose. For more clarification I have added a comment too.

Reply:
thanks for that I did a bit of research my self and it's common for Arduino to be in loops thanks for the clarification.